Finish making the defense section more approachable (#46)
xsleaks/wiki: content/docs/defenses/secure-defaults/corb.md

menu = "main"
+++

Cross-Origin Read Blocking (CORB) is a security mechanism that prevents attackers from loading certain cross-origin resources [^1]. This protection was created to defend against speculative side-channel attacks such as Spectre that allow attackers to read the memory of their own process. CORB aims to prevent attackers from loading certain sensitive cross-origin resources into an attacker controlled process. For example, if the attacker tries to load cross-origin HTML, XML, or JSON into an `img` tag, CORB will prevent this from happening. With CORB, it will be treated as though the server returned no data. 

> CORB is important both for defending against XS-Leaks and for defending against Spectre attacks. 

Author: This summary was confusing to me, so I think it is better to cut it and then just let people go to the official docs on CORB. Especially since CORB has changed over time and is already very complex.

> CORB is important both for defending against XS-Leaks and for defending against Spectre attacks. 

Reviewer: ```suggestion
CORB is important for defending against XS-Leaks, especially Spectre-alike attacks. 
```

Spectre is also an XS-Leak

Author: Changed to "Spectre-like" rather than "Spectre-alike" but :+1:

> To classify resources, CORB uses the `Content-Type` header, the `nosniff` header, and a variety of other heuristics. 

Author: This summary was confusing to me, so I think it is better to cut it and then just let people go to the official docs on CORB. Especially since CORB has changed over time and is already very complex.

Reviewer: Agree. 

Maybe something around: `Read more how default CORB heuristics work`
